src/components/TodoItem.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo,
+  loadingId: number[],
+  isLoaderActive: boolean,
+  handleComplete: (arg: Todo) => void,
+  handleEditTodo: (updatedTodo: Todo) => Promise<void>,
+  handleDeleteTodo: (id: number) => Promise<void>,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  loadingId,
+  isLoaderActive,
+  handleComplete,
+  handleEditTodo,
+  handleDeleteTodo,
+}) => {
+  const {
+    id, completed, title, userId,
+  } = todo;
+
+  const [editedTitle, setEditedTitle] = useState(title);
+  const [isUpdateFormActive, setIsUpdateFormActive] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (isUpdateFormActive) {
+      inputRef.current?.focus();
+    }
+  }, [isUpdateFormActive]);
+
+  const handleDoubleClick = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (event.detail === 2) {
+      setIsUpdateFormActive(true);
+    }
+  };
+
+  const submit = async () => {
+    if (editedTitle.trim()) {
+      await handleEditTodo({
+        id, title: editedTitle, userId, completed,
+      });
+    } else {
+      await handleDeleteTodo(id);
+    }
```

Comment:
API calls with `await` have to be wrapped with `try/catch` block to handle errors